source/Ship.h

```diff
@@ -425,6 +426,9 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// Add or remove a ship from this ship's list of escorts.
 	void AddEscort(Ship &ship);
 	void RemoveEscort(const Ship &ship);
+	// Cache relevant data for all escorts or the given escort.
+	void TuneForEscorts();
+	void TuneForEscort(std::shared_ptr<Ship> &ship);
```

Comment:
```suggestion
	void TuneForEscort(const std::shared_ptr<Ship> &ship);
```
The given ship isn't modified